### PR TITLE
added directory Bakes

### PR DESCRIPTION
**Pull request by gtg092x**
@kyleburton - I thought this was a good way to get introduced to your work!

This allows Bakefile to be a directory. When the user does this, all files in the directory are treated as `Bakefile`s.

### CODE REVIEW

**Comment by kyleburton** [comment]
I think I like the 'automagic' of using a directory (as you indicated in person is a feature of rake).  It hits the goal of eliminating steps the user has to take (writing the require statements) with only a small amount of additional a priori knowledge.

Can we think a bit about the name of that subdirectory?  Do you feel it's clear that the `Bakefile` could be either a file or directory?  

What suggestions do you have?

* `tasks` same as rake, doesn't immediately say 'bake' to the user, may conflict with other tools?
* `.bake` correlates with bake, is a hidden dir obvious to users?
* `baketasks` correlates with bake, kinda wordy

Do you have any other suggestions?

File: bake [hunk 1]
```diff
@@ -331,17 +331,26 @@ function bake_cd () {
...
+  if [ -d "bake" ]; then
+    echo "$project_root/bake"
+    return 0
+  fi
+
+  if [ -d "Bake" ]; then
+    echo "$project_root/Bake"
+    return 0
+  fi
+
```
Comment (gtg092x) [inline bake]: @kyleburton here's where I'm setting the bake directory

File: bake [hunk 2]
```diff
@@ -331,17 +331,26 @@ function bake_cd () {
...
+  if [ -f "Bakefile" ]; then
     echo "$project_root/Bakefile"
     return 0
   fi
 
-  if [ -e "bakefile" ]; then
+  if [ -f "bakefile" ]; then
     echo "$project_root/bakefile"
     return 0
   fi
```
Comment (gtg092x) [inline bake]: @kyleburton - this change forces `bakefile` to be a file (I think). Trying to force a differentiation from the bake dir.

**Comment by kyleburton** [comment]
I think this all looks good, for shellcheck, I'm primarily concerned about bake itself so the tests having a few issues is ok w/me.  thanks!

**Pull request merged**